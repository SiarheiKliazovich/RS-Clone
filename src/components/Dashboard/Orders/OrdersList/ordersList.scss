@import "../../../../sass/variables";
@import "../../../../sass/mixins";

.orders-list__wrapper {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
}

.orders-list__item {
    width: 100%;

    table {
        width: 100%;

        th {
            height: 35px;
            vertical-align: middle;

            &:first-child {
                width: 40px;
            }

            @include media-mobile {

                &:nth-child(2) {
                    display: none;
                }

                &:nth-child(3) {
                    width: 40px;
                }
            }

            @include media-small-mobile {
                
            &:first-child {
                width: 20px;
            }
                &:nth-child(3) {
                    display: none;
                }

            }
        }
        tr {
            vertical-align: middle;
        }
    }
}

.select {

    @include media-mobile {
        max-width: 120px;    
    }  

    @include media-small-mobile {
        max-width: 80px;    
    }  
    
}

.icon {
    background-image: url("../../../../assets/images/png/icon-1.png");
    background-size: contain;
    background-repeat: no-repeat;
    width: 30px;
    height: 35px;
    border: none;
    cursor: pointer;

    @include media-small-mobile {
        width: 20px; 
        background-position: center;
    }
}

.button-apply {
    cursor: pointer;
    border-radius: 5px;
}
