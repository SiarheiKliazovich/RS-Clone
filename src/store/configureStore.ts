/* eslint-disable @typescript-eslint/ban-ts-comment */
import { applyMiddleware, createStore } from "redux";
import thunkMiddleware from "redux-thunk";

import loggerMiddleware from "./logger";
import rootReducer from "../reducers";

const configureStore = () => {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, middlewareEnhancer);

  // @ts-ignore:next-line
  if (process.env.NODE_ENV !== "production" && module.hot) {
    // @ts-ignore:next-line
    module.hot.accept("./../reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
};

export default configureStore;
