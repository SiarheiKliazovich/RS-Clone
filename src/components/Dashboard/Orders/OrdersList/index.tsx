import { FunctionComponent } from "react";
import { useDispatch } from "react-redux";

import {
  deleteOrder,
  getOrders,
  updateOrder,
} from "../../../../services/apiOrders";
import {
  orderDeleted,
  ordersFetching,
  ordersFetched,
  ordersFetchingError,
} from "../../../../actions";
import { IOrdersListComponent } from "../../../../interfaces/ordersListComponent";
import OrdersListItem from "../OrdersListItem";
import { useTranslation } from "react-i18next";

import "./ordersList.scss";

const OrdersList: FunctionComponent<IOrdersListComponent> = ({ orders }) => {
  const dispatch = useDispatch();

  const { t } = useTranslation("dashboard");

  const deleteItem = async (id: string) => {
    try {
      await deleteOrder(id);
      dispatch(orderDeleted(id));
      dispatch(ordersFetching());
      const orders = await getOrders();
      dispatch(ordersFetched(orders));
    } catch {
      dispatch(ordersFetchingError());
    }
  };

  const updateItem = async (id: string, status: string) => {
    try {
      await updateOrder(id, { status: status });
      dispatch(ordersFetching());
      const orders = await getOrders();
      dispatch(ordersFetched(orders));
    } catch {
      dispatch(ordersFetchingError());
    }
  };

  return (
    <>
      <div className="orders-list__wrapper">
        <div className="orders-list__item">
          <table>
            <tbody>
              <tr className="table-header">
                <th>#</th>
                <th>{t("ordersTable.date")}</th>
                <th>{t("ordersTable.price")}</th>
                <th>{t("ordersTable.status")}</th>
              </tr>
              {orders.map((item, i) => {
                return (
                  <OrdersListItem
                    key={item._id}
                    item={item}
                    i={i}
                    deleteItem={deleteItem}
                    updateItem={updateItem}
                  />
                );
              })}
            </tbody>
          </table>
        </div>
      </div>
    </>
  );
};

export default OrdersList;
