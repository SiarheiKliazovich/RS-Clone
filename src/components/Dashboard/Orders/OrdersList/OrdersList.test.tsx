import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import OrdersList from ".";

const mockStore = configureStore([]);

describe("OrdersList", () => {
  // eslint-disable-next-line
  let store: any;

  beforeEach(() => {
    store = mockStore({
      appInterface: {
        accentColor: {
          static: "#000",
          hover: "#fff",
        },
      },
    });
  });

  test("renders correctly", () => {
    const orders = [
      {
        _id: "123",
        date: "2022-02-24T10:00:00.000Z",
        price: 100,
        status: "PAID",
        userId: "userId",
        sellerId: "sellerId",
        image: "image",
      },
      {
        _id: "456",
        date: "2022-02-24T11:00:00.000Z",
        price: 200,
        status: "FIHISHED",
        userId: "userId",
        sellerId: "sellerId",
        image: "image",
      },
    ];
    const { container } = render(
      <Provider store={store}>
        <Router>
          <OrdersList orders={orders} />
        </Router>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
