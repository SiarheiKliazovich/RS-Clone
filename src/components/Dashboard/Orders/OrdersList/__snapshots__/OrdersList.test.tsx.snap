// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrdersList renders correctly 1`] = `
<div
  class="orders-list__wrapper"
>
  <div
    class="orders-list__item"
  >
    <table>
      <tbody>
        <tr
          class="table-header"
        >
          <th>
            #
          </th>
          <th>
            ordersTable.date
          </th>
          <th>
            ordersTable.price
          </th>
          <th>
            ordersTable.status
          </th>
        </tr>
        <tr>
          <th>
            1
          </th>
          <th>
            2022-02-24
          </th>
          <th>
            100
          </th>
          <th>
            PAID
          </th>
          <th>
            <button
              class="button__delete-order"
            >
              <i
                class="fa fa-remove"
              />
              ordersTable.delete
            </button>
          </th>
          <th
            class="table-select"
          >
            <select
              class="select"
              style="color: rgb(43, 52, 59);"
            >
              <option
                disabled=""
                hidden=""
                value="EMPTY"
              >
                select.change
              </option>
              <option
                value="NON-PAID"
              >
                select.nonPaid
              </option>
              <option
                value="PAID"
              >
                select.paid
              </option>
              <option
                value="DECLINED"
              >
                select.declined
              </option>
              <option
                value="IN-PROGRESS"
              >
                select.inProgress
              </option>
              <option
                value="FINISHED"
              >
                select.finished
              </option>
            </select>
          </th>
          <th>
            <button
              class="button-apply"
              style="background-color: rgb(0, 0, 0);"
            >
              ordersTable.apply
            </button>
          </th>
        </tr>
        <tr>
          <th>
            2
          </th>
          <th>
            2022-02-24
          </th>
          <th>
            200
          </th>
          <th>
            FIHISHED
          </th>
          <th>
            <button
              class="button__delete-order"
            >
              <i
                class="fa fa-remove"
              />
              ordersTable.delete
            </button>
          </th>
          <th
            class="table-select"
          >
            <select
              class="select"
              style="color: rgb(43, 52, 59);"
            >
              <option
                disabled=""
                hidden=""
                value="EMPTY"
              >
                select.change
              </option>
              <option
                value="NON-PAID"
              >
                select.nonPaid
              </option>
              <option
                value="PAID"
              >
                select.paid
              </option>
              <option
                value="DECLINED"
              >
                select.declined
              </option>
              <option
                value="IN-PROGRESS"
              >
                select.inProgress
              </option>
              <option
                value="FINISHED"
              >
                select.finished
              </option>
            </select>
          </th>
          <th>
            <button
              class="button-apply"
              style="background-color: rgb(0, 0, 0);"
            >
              ordersTable.apply
            </button>
          </th>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
